### ISSUE TITLE
Packaging Chain-specific Configs & Effects

### ISSUE THREAD

**Issue by harrysolovay** [paritytech/capi#213]
How should we approach packaging chain-specific configs and effects?

## Current approach

Currently, we expose several configs––each with the same tmp types––from the `known` dir. DNT takes care of exposing this directory from the generated NPM package's `exports`. Assuming that a given Node project is configured with Node16/NodeNext import resolution, its devs can do the following.

```ts
import { polkadot } from "capi/known";
```

However, this opens a can of worms depending on adjacent config (especially in front-end frameworks such as NextJS) cc @karl-kallavus. 

## Alternative

We could create some means of enabling the community to publish to the `@capi` org on NPM (similarly to `@types`). The maintainers of a given chain can take ownership over its `@capi`-namespaced package. This is in line with the AWS CDK's treatment of per-service constructs.

```ts
import * as cdk from "@aws-cdk/core";
import * as lambda from "@aws-cdk/aws-lambda";
```

In our case, `@capi`-namespaced packages would contain configs and effects that abstract over the given chain.

```ts
import * as C from "capi";
import { polkadot } from "@capi/polkadot";
```

The drawback of this approach is that the developer must install an additional library for each chain. However, this seems unavoidable in the long term. We certainly would not want to package the specifics of dozens of chains in the Capi repo.

## Closing Thoughts

We need to decide on this separation of concerns: IMO, the Capi repo should need not contain chain-specific values and types; it should contain only the means of representing (and––where possible––generating) those chain-specifics.

**Comment by tjjfvi** [comment]
> We could create some means of enabling the community to publish to the `@capi` org on NPM (similarly to `@types`).

I've observed the amount of work done to maintain DT, and that kind of publishing is a *huge* undertaking.

To any degree significant chain-specific packages are needed, I think it can be covered by people publishing `capi-xyz` packages. We may also want to publish some `@capi/xyz` packages that are "official" – for example, `@capi/polkadot`.

However, I think it would be best if we can minimize the degree to which this is needed. Ideally, as much of this comes through codegen as possible. If we can get the manual chain-specific metadata down to, say, a single short file, at that point I think it would make the most sense to simply have a collection of those files in the `capi` repo directly.

**Comment by harrysolovay** [comment]
Perhaps closed by #226, should we all agree on new direction.
